datasets/nq_open/nq_open.py

```diff
@@ -0,0 +1,133 @@
+    booktitle = "Proceedings of the 57th Annual Meeting of the Association for Computational Linguistics",
+    month = jul,
+    year = "2019",
+    address = "Florence, Italy",
+    publisher = "Association for Computational Linguistics",
+    url = "https://www.aclweb.org/anthology/P19-1612",
+    doi = "10.18653/v1/P19-1612",
+    pages = "6086--6096",
+    abstract = "Recent work on open domain question answering (QA) assumes strong supervision of the supporting evidence and/or assumes a blackbox information retrieval (IR) system to retrieve evidence candidates. We argue that both are suboptimal, since gold evidence is not always available, and QA is fundamentally different from IR. We show for the first time that it is possible to jointly learn the retriever and reader from question-answer string pairs and without any IR system. In this setting, evidence retrieval from all of Wikipedia is treated as a latent variable. Since this is impractical to learn from scratch, we pre-train the retriever with an Inverse Cloze Task. We evaluate on open versions of five QA datasets. On datasets where the questioner already knows the answer, a traditional IR system such as BM25 is sufficient. On datasets where a user is genuinely seeking an answer, we show that learned retrieval is crucial, outperforming BM25 by up to 19 points in exact match.",
+}
+"""
+
+_URLS = {
+    "dev": "https://raw.githubusercontent.com/google-research-datasets/natural-questions/master/nq_open/NQ-open.efficientqa.dev.1.1.jsonl",
+    "train": "https://raw.githubusercontent.com/google-research-datasets/natural-questions/master/nq_open/NQ-open.train.jsonl",
+}
+
+
+class NQOpenV2Config(datasets.BuilderConfig):
+    """BuilderConfig for NQ_Open."""
+
+    def __init__(self, **kwargs):
+        """BuilderConfig for NQ_Open.
+        Args:
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(NQOpenV2Config, self).__init__(**kwargs)
+
+
+class NQOpen(datasets.GeneratorBasedBuilder):
+    """ NQ_Open open domain question answering dataset. """
+
+    BUILDER_CONFIGS = [
+        NQOpenV2Config(
+            name="nq_open",
+            version=datasets.Version("1.0.0", ""),
+            description=f"NQ_Open open domain question answering dataset.",
+        ),
+    ]
+    BUILDER_CONFIG_CLASS = NQOpenV2Config
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "question": datasets.Value("string"),
+                    "answer": datasets.Sequence(datasets.Value("string")),
+                }
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE_URL,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        train_path = dl_manager.download_and_extract(_URLS["train"])
+        val_path = dl_manager.download_and_extract(_URLS["dev"])
```

Comment:
`dl_manager.download_and_extract` can take a dict of URLs as input, returning a dict of paths:
```
paths = dl_manager.download_and_extract(_URLS)
```

Reply:
okay will update it